fury/actor.py

```diff
@@ -1204,20 +1212,21 @@ def _tensor_slicer_mapper(evals, evecs, affine=None, mask=None, sphere=None,
     all_xyz = np.ascontiguousarray(np.concatenate(all_xyz))
     all_xyz_vtk = numpy_support.numpy_to_vtk(all_xyz, deep=True)
 
-    all_faces = np.concatenate(all_faces)
-    all_faces = np.hstack((3 * np.ones((len(all_faces), 1)),
-                           all_faces))
-    ncells = len(all_faces)
-
-    all_faces = np.ascontiguousarray(all_faces.ravel(), dtype='i8')
-    all_faces_vtk = numpy_support.numpy_to_vtkIdTypeArray(all_faces,
-                                                          deep=True)
-
     points = vtk.vtkPoints()
     points.SetData(all_xyz_vtk)
 
+    all_faces = np.concatenate(all_faces)
     cells = vtk.vtkCellArray()
-    cells.SetCells(ncells, all_faces_vtk)
+    if vtk.vtkVersion.GetVTKMajorVersion() >= 9:
+        cells = numpy_to_vtk_cells(all_faces, is_coords=False)
+    else:
+        all_faces = np.hstack((3 * np.ones((len(all_faces), 1)),
+                              all_faces))
+        ncells = len(all_faces)
+        all_faces = np.ascontiguousarray(all_faces.ravel(), dtype='i8')
+        all_faces_vtk = numpy_support.numpy_to_vtkIdTypeArray(all_faces,
+                                                              deep=True)
+        cells.SetCells(ncells, all_faces_vtk)
```

Comment:
Might be better to make a function out of this, so you don't have to repeat this code. 

The easiest for maintenance in the future would probably be to make one function that encompasses the different aspects handled in this PR, but that might end up a bit of a multi-headed monster.

Reply:
I agree. In general, `tensor_slicer` and `odf_slicer`function need some refactoring. A lot of code is duplicated.

To answer your proposition, This is why I created `numpy_to_vtk_cells` function but it seems it needs a bit more thinking to manage more cases.  I will update that.

Thank you for the quick review!